File: Omega/src/com/saggitt/omega/preferences/ThemedPreferenceCategory.java
/*
 *  This file is part of Omega Launcher.
 *  Copyright (c) 2021   Saul Henriquez
 *
 *  This program is free software: you can redistribute it and/or modify
 *  it under the terms of the GNU General Public License as
 *  published by the Free Software Foundation, either version 3 of the
 *  License, or (at your option) any later version.
 *
 *  This program is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *  GNU General Public License for more details.
 *
 *  You should have received a copy of the GNU General Public License
 *  along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */

package com.saggitt.omega.preferences;

import android.content.Context;
import android.util.AttributeSet;
import android.widget.TextView;

import androidx.preference.PreferenceCategory;
import androidx.preference.PreferenceViewHolder;

import com.android.launcher3.Utilities;

public class ThemedPreferenceCategory extends PreferenceCategory {
    private Context mContext;

    public ThemedPreferenceCategory(Context context) {
        this(context, null, 0);
    }
    public ThemedPreferenceCategory(Context context, AttributeSet attrs) {
        this(context, attrs, 0);
    }

    public ThemedPreferenceCategory(Context context, AttributeSet attrs, int defStyle) {
        super(context, attrs, defStyle);
        mContext = context;
    }

    @Override
    public void onBindViewHolder(PreferenceViewHolder holder) {
        super.onBindViewHolder(holder);
        TextView title = (TextView) holder.findViewById(android.R.id.title);
        title.setTextSize(16f);
        title.setTextColor(Utilities.getOmegaPrefs(mContext).getAccentColor());
    }
}

File: Omega/src/com/saggitt/omega/preferences/ThemedPreferenceCategory.kt
/*
 *  This file is part of Omega Launcher.
 *  Copyright (c) 2021   Saul Henriquez
 *
 *  This program is free software: you can redistribute it and/or modify
 *  it under the terms of the GNU General Public License as
 *  published by the Free Software Foundation, either version 3 of the
 *  License, or (at your option) any later version.
 *
 *  This program is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *  GNU General Public License for more details.
 *
 *  You should have received a copy of the GNU General Public License
 *  along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */
package com.saggitt.omega.preferences

import android.R.id
import android.content.Context
import android.util.AttributeSet
import android.widget.TextView
import androidx.preference.PreferenceCategory
import androidx.preference.PreferenceViewHolder
import com.android.launcher3.Utilities

class ThemedPreferenceCategory @JvmOverloads constructor(
    private val mContext: Context,
    attrs: AttributeSet? = null, defStyle: Int = 0
) : PreferenceCategory(mContext, attrs, defStyle) {

    override fun onBindViewHolder(holder: PreferenceViewHolder) {
        super.onBindViewHolder(holder)
        val title = holder.findViewById(id.title) as TextView
        title.textSize = 16f
        title.setTextColor(Utilities.getOmegaPrefs(mContext).accentColor)
    }
}